bgpd/bgp_label.c

```diff
@@ -119,21 +119,137 @@ mpls_label_t bgp_adv_label(struct bgp_node *rn, struct bgp_path_info *pi,
 
 	return rn->local_label;
 }
+/**
+ * This is passed as the callback function to bgp_labelpool.c:bgp_lp_get()
+ * by bgp_reg_dereg_for_label() when a label needs to be obtained from
+ * label pool.
+ * Note that it will reject the allocated label if a label index is found,
+ * because the label index supposes predictable labels
+ */
+int bgp_reg_for_label_callback(
+	mpls_label_t new_label,
+	void *labelid,
+	bool allocated)
+{
+	struct bgp_path_info *pi = (struct bgp_path_info *)labelid;
+	struct bgp_node *rn = (struct bgp_node *)pi->net;
+	char addr[PREFIX_STRLEN];
+
+	prefix2str(&rn->p, addr, PREFIX_STRLEN);
+
+	zlog_debug("%s: FEC %s label=%u, allocated=%d",
+			__func__, addr, new_label, allocated);
```

Comment:
This needs to debug guard of some kind.  See bgp_labelpool.c for an example.

Reply:
I have decided to use the same debug guard as in bgp_labelpool.c (BGP_DEBUG(labelpool, LABELPOOL)) as bgp_reg_for_label_callback() will only be called when we rely on label pool to assign labels to prefixes. So, it did not make sense to me to create another debug item. Will push fixes shortly.